kiwi/system/setup.py

```diff
@@ -302,8 +299,10 @@ def setup_users(self):
                 user_id = user.get_id()
                 user_realname = user.get_realname()
                 user_shell = user.get_shell()
-                user_secgroups = user.get_secgroups()
-                user_group = user.get_group()
+                if user.get_groups():
+                    user_groups = user.get_groups().split(',')
+                else:
+                    user_groups = None
```

Comment:
This "hard to read" is more than excuse everywhere. It is not about reading the code, but _understanding_ it. In both cases, if you do:
```python
if foo is not None:
   result = foo[0]
else:
   result = foo
```
...or:
```python
result = foo[0] if foo is not None else foo
```

In both cases I would still question why you would wrote that bug? :wink: Because in both cases you can read, that `foo` can be also `[]`.